In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 5GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/ipldata/deliveries.csv
/kaggle/input/ipldata/matches.csv


In [5]:
df_match = pd.read_csv("/kaggle/input/ipldata/matches.csv")
df_match.head()

id  season       city        date                        team1  \
0   1    2017  Hyderabad  2017-04-05          Sunrisers Hyderabad   
1   2    2017       Pune  2017-04-06               Mumbai Indians   
2   3    2017     Rajkot  2017-04-07                Gujarat Lions   
3   4    2017     Indore  2017-04-08       Rising Pune Supergiant   
4   5    2017  Bangalore  2017-04-08  Royal Challengers Bangalore   

                         team2                  toss_winner toss_decision  \
0  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
1       Rising Pune Supergiant       Rising Pune Supergiant         field   
2        Kolkata Knight Riders        Kolkata Knight Riders         field   
3              Kings XI Punjab              Kings XI Punjab         field   
4             Delhi Daredevils  Royal Challengers Bangalore           bat   

   result  dl_applied                       winner  win_by_runs  \
0  normal           0          Sunrisers Hyderabad           35   
1  normal           0       Rising Pune Supergiant            0   
2  normal           0        Kolkata Knight Riders            0   
3  normal           0              Kings XI Punjab            0   
4  normal           0  Royal Challengers Bangalore           15   

   win_by_wickets player_of_match                                      venue  \
0               0    Yuvraj Singh  Rajiv Gandhi International Stadium, Uppal   
1               7       SPD Smith    Maharashtra Cricket Association Stadium   
2              10         CA Lynn     Saurashtra Cricket Association Stadium   
3               6      GJ Maxwell                     Holkar Cricket Stadium   
4               0       KM Jadhav                      M Chinnaswamy Stadium   

          umpire1        umpire2 umpire3  
0     AY Dandekar       NJ Llong     NaN  
1  A Nand Kishore         S Ravi     NaN  
2     Nitin Menon      CK Nandan     NaN  
3    AK Chaudhary  C Shamshuddin     NaN  
4             NaN            NaN     NaN

In [10]:
df_match.winner.mode()
df_match.toss_winner.mode()

0    Mumbai Indians
dtype: object

In [21]:
df_abu = df_match[df_match.venue == "Sheikh Zayed Stadium"]
df_abu

id  season       city        date                        team1  \
457  458    2014  Abu Dhabi  2014-04-16        Kolkata Knight Riders   
459  460    2014  Abu Dhabi  2014-04-18          Chennai Super Kings   
460  461    2014  Abu Dhabi  2014-04-18          Sunrisers Hyderabad   
464  465    2014  Abu Dhabi  2014-04-21          Chennai Super Kings   
470  471    2014  Abu Dhabi  2014-04-26  Royal Challengers Bangalore   
471  472    2014  Abu Dhabi  2014-04-26              Kings XI Punjab   
475  476    2014  Abu Dhabi  2014-04-29             Rajasthan Royals   

                     team2            toss_winner toss_decision  result  \
457         Mumbai Indians  Kolkata Knight Riders           bat  normal   
459        Kings XI Punjab    Chennai Super Kings           bat  normal   
460       Rajasthan Royals       Rajasthan Royals         field  normal   
464       Delhi Daredevils    Chennai Super Kings           bat  normal   
470       Rajasthan Royals       Rajasthan Royals         field  normal   
471  Kolkata Knight Riders  Kolkata Knight Riders         field  normal   
475  Kolkata Knight Riders       Rajasthan Royals           bat     tie   

     dl_applied                 winner  win_by_runs  win_by_wickets  \
457           0  Kolkata Knight Riders           41               0   
459           0        Kings XI Punjab            0               6   
460           0       Rajasthan Royals            0               4   
464           0    Chennai Super Kings           93               0   
470           0       Rajasthan Royals            0               6   
471           0        Kings XI Punjab           23               0   
475           0       Rajasthan Royals            0               0   

    player_of_match                 venue          umpire1         umpire2  \
457       JH Kallis  Sheikh Zayed Stadium        M Erasmus  RK Illingworth   
459      GJ Maxwell  Sheikh Zayed Stadium   RK Illingworth   C Shamshuddin   
460       AM Rahane  Sheikh Zayed Stadium        BF Bowden  RK Illingworth   
464        SK Raina  Sheikh Zayed Stadium   RK Illingworth   C Shamshuddin   
470        PV Tambe  Sheikh Zayed Stadium  HDPK Dharmasena   C Shamshuddin   
471  Sandeep Sharma  Sheikh Zayed Stadium  HDPK Dharmasena  RK Illingworth   
475     JP Faulkner  Sheikh Zayed Stadium        Aleem Dar    AK Chaudhary   

    umpire3  
457     NaN  
459     NaN  
460     NaN  
464     NaN  
470     NaN  
471     NaN  
475     NaN

In [18]:
df_abu = df_match[df_match.venue == "Sheikh Zayed Stadium"]
df_abu.loc[[457,464]]

id  season       city        date                  team1  \
457  458    2014  Abu Dhabi  2014-04-16  Kolkata Knight Riders   
464  465    2014  Abu Dhabi  2014-04-21    Chennai Super Kings   

                team2            toss_winner toss_decision  result  \
457    Mumbai Indians  Kolkata Knight Riders           bat  normal   
464  Delhi Daredevils    Chennai Super Kings           bat  normal   

     dl_applied                 winner  win_by_runs  win_by_wickets  \
457           0  Kolkata Knight Riders           41               0   
464           0    Chennai Super Kings           93               0   

    player_of_match                 venue         umpire1         umpire2  \
457       JH Kallis  Sheikh Zayed Stadium       M Erasmus  RK Illingworth   
464        SK Raina  Sheikh Zayed Stadium  RK Illingworth   C Shamshuddin   

    umpire3  
457     NaN  
464     NaN

From the past results we can see that Mumbai is a tough performer and has a higher probability of winning the game at Abu Dhabi.

As past data says that the toss winner most of the times choose to bat, I will choose the answer bat.

In [24]:
df_rohit = pd.read_html("https://www.iplt20.com/teams/mumbai-indians/squad/107/rohit-sharma")
Batting = df_rohit[2]
Batting

Batting and Fielding  Mat  No  Runs    HS    Ave    BF      SR  100  50  \
0                Career  194  28  5109  109*  31.73  3889  131.37    1  38   
1                  2020    6   0   211    80  35.16   145  145.51    0   2   
2                  2019   15   1   405    67  28.92   315  128.57    0   2   
3                  2018   14   2   286    94  23.83   215  133.02    0   2   
4                  2017   17   2   333    67  23.78   273  121.97    0   3   
5                  2016   14   3   489   85*  44.45   368  132.88    0   5   
6                  2015   16   2   482   98*  34.42   333  144.74    0   3   
7                  2014   15   2   390   59*  30.00   302  129.13    0   3   
8                  2013   19   5   538   79*  38.42   409  131.54    0   4   
9                  2012   17   2   433  109*  30.92   342  126.60    1   3   
10                 2011   16   3   372    87  33.81   297  125.25    0   3   
11                 2010   16   2   404    73  28.85   302  133.77    0   3   
12                 2009   16   3   362    52  27.84   315  114.92    0   1   
13                 2008   13   1   404   76*  36.72   273  147.98    0   4   

     4s   6s  CT  ST  
0   446  208  87   0  
1    15   14   4   0  
2    52   10   4   0  
3    25   12   8   0  
4    31    9  10   0  
5    49   16   2   0  
6    41   21   5   0  
7    31   16   5   0  
8    35   28   7   0  
9    39   18  13   0  
10   32   13   7   0  
11   36   14   9   0  
12   22   18   5   0  
13   38   19   8   0

In [29]:
Batting["Runs"][1]/Batting["Mat"][1]

35.166666666666664

Looks like he will score around 30-40

In [6]:
df_delivery = pd.read_csv("/kaggle/input/ipldata/deliveries.csv")
df_delivery.head()

match_id  inning         batting_team                 bowling_team  over  \
0         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
1         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
2         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
3         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
4         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   

   ball    batsman non_striker    bowler  is_super_over  ...  bye_runs  \
0     1  DA Warner    S Dhawan  TS Mills              0  ...         0   
1     2  DA Warner    S Dhawan  TS Mills              0  ...         0   
2     3  DA Warner    S Dhawan  TS Mills              0  ...         0   
3     4  DA Warner    S Dhawan  TS Mills              0  ...         0   
4     5  DA Warner    S Dhawan  TS Mills              0  ...         0   

   legbye_runs  noball_runs  penalty_runs  batsman_runs  extra_runs  \
0            0            0             0             0           0   
1            0            0             0             0           0   
2            0            0             0             4           0   
3            0            0             0             0           0   
4            0            0             0             0           2   

   total_runs  player_dismissed dismissal_kind fielder  
0           0               NaN            NaN     NaN  
1           0               NaN            NaN     NaN  
2           4               NaN            NaN     NaN  
3           0               NaN            NaN     NaN  
4           2               NaN            NaN     NaN  

[5 rows x 21 columns]

In [30]:
df_delivery.columns

Index(['match_id', 'inning', 'batting_team', 'bowling_team', 'over', 'ball',
       'batsman', 'non_striker', 'bowler', 'is_super_over', 'wide_runs',
       'bye_runs', 'legbye_runs', 'noball_runs', 'penalty_runs',
       'batsman_runs', 'extra_runs', 'total_runs', 'player_dismissed',
       'dismissal_kind', 'fielder'],
      dtype='object')

In [42]:
df_delivery[df_delivery["bowler"] == "(.*)"]

0         False
1         False
2         False
3         False
4         False
          ...  
179073    False
179074    False
179075    False
179076    False
179077    False
Name: bowler, Length: 179078, dtype: bool

In [44]:
df_rabada = pd.read_html("https://www.iplt20.com/teams/delhi-capitals/squad/1664/kagiso-rabada")
df_rabada = df_rabada[3]

In [45]:
df_rabada

Bowling  Mat  Balls  Runs  WKTS   BBM    Ave  Econ     SR  4W  5W
0  Career   24    552   741    46  4/21  16.10  8.05  12.00   3   0
1    2020    6    142   185    15  4/24  12.33  7.81   9.46   1   0
2    2019   12    282   368    25  4/21  14.72  7.82  11.28   2   0
3    2017    6    128   188     6  2/20  31.33  8.81  21.33   0   0

In [48]:
df_rabada.index = df_rabada.Bowling
df_rabada.drop("Bowling",axis=1, inplace=True)

In [53]:
df_rabada
df_rabada["WKTS"][1]/df_rabada["Mat"][1]

2.5

Out of 24 matches he just get a one 4w in and average wickets taken is 2.5..